mk/config.mk

```diff
@@ -984,13 +984,16 @@ endif
 CFG_WDT ?= n
 
 # Enable watchdog SMC handling compatible with arm-smc-wdt Linux driver
-# When enabled, CFG_WDT_SM_HANDLER_ID must be defined with a SMC ID
 CFG_WDT_SM_HANDLER ?= n
 
 $(eval $(call cfg-enable-all-depends,CFG_WDT_SM_HANDLER,CFG_WDT))
-ifeq (y-,$(CFG_WDT_SM_HANDLER)-$(CFG_WDT_SM_HANDLER_ID))
-$(error CFG_WDT_SM_HANDLER_ID must be defined when enabling CFG_WDT_SM_HANDLER)
-endif
+
+# When CFG_WDT_SM_HANDLER=y and OP-TEE is used as monitor,
+# SMC function ID 0x82003D06 implements arm-smc-wdt service.
+# Platform can also define an additional platform specific SMC function ID
```

Comment:
Proposal for rephrasing:
\# When CFG_WDT_SM_HANDLER=y and OP-TEE is used as monitor **(i.e. OP-TEE 32bit Secure Monitor)**
\# SMC function ID 0x82003D06 **default** implements arm-smc-wdt service.
\# Platform can **~~also define an additional~~ override this ID with a** platform specific SMC function ID
\# to access arm-smc-wdt service thanks to optional config switch
\# CFG_WDT_SM_HANDLER_ID.

Reply:
Agree

Reply:
@jenswi-linaro Any comment before I update and squash ?

Reply:
Please go ahead and squash.